zebrad/src/commands/migrate_from_zcashd.rs

```diff
@@ -0,0 +1,371 @@
+                        source_block.clone().into(),
+                    )
+                })
+                .await?;
+            let target_block_commit_hash = match target_block_commit_hash {
+                zebra_state::Response::Committed(target_block_commit_hash) => {
+                    trace!(?target_block_commit_hash, "wrote Zebra block");
+                    target_block_commit_hash
+                }
+                response => Err(format!(
+                    "unexpected response to CommitSemanticallyVerifiedBlock request, height: {}\n \
+                     response: {response:?}",
+                    height.0,
+                ))?,
+            };
+
+            // Read written block from target
+            let target_block = target_state
+                .ready()
+                .await?
+                .call(zebra_state::Request::Block(height.into()))
+                .await?;
+            let target_block = match target_block {
+                zebra_state::Response::Block(Some(target_block)) => {
+                    trace!(?height, %target_block, "read Zebra block");
+                    target_block
+                }
+                zebra_state::Response::Block(None) => Err(format!(
+                    "unexpected missing Zebra block, height: {}",
+                    height.0,
+                ))?,
+
+                response => Err(format!(
+                    "unexpected response to Block request, height: {},\n \
+                     response: {response:?}",
+                    height.0,
+                ))?,
+            };
+            let target_block_data_hash = target_block.hash();
+
+            // Check for data errors
+            //
+            // These checks make sure that Zebra doesn't corrupt the block data
+            // when serializing it.
+            // Zebra currently serializes `Block` structs into bytes while writing,
+            // then deserializes bytes into new `Block` structs when reading.
+            // So these checks are sufficient to detect block data corruption.
+            //
+            // If Zebra starts reusing cached `Block` structs after writing them,
+            // we'll also need to check `Block` structs created from the actual database bytes.
+            if source_block_hash != target_block_commit_hash
+                || source_block_hash != target_block_data_hash
+                || source_block != target_block
+            {
+                Err(format!(
+                    "unexpected mismatch between zcashd and Zebra blocks,\n \
+                     max copy height: {max_copy_height:?},\n \
+                     zcashd hash: {source_block_hash:?},\n \
+                     Zebra commit hash: {target_block_commit_hash:?},\n \
+                     Zebra data hash: {target_block_data_hash:?},\n \
+                     zcashd block: {source_block:?},\n \
+                     Zebra block: {target_block:?}",
+                ))?;
+            }
```

Comment:
The first two checks here:

```rust
            if source_block_hash != target_block_commit_hash
                || source_block_hash != target_block_data_hash
                || source_block != target_block
```

are now happening [here in the block verifier](https://github.com/str4d/zebra/blob/cli-migrate-from-zcashd/zebra-consensus/src/block.rs#L369) and [here in the checkpoint verifier](https://github.com/str4d/zebra/blob/cli-migrate-from-zcashd/zebra-consensus/src/checkpoint.rs#L1137). 

Is the third check necessary?

```suggestion
            block_verifier_router
                .ready()
                .await?
                .call(zebra_consensus::Request::Commit(source_block))
                .await?;
```